packages/template-retail-react-app/app/components/experience/image-with-text/index.jsx

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright (c) 2023, Salesforce, Inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import React, {useEffect, useState} from 'react'
+import PropTypes from 'prop-types'
+import {Image, Box, Text, Link} from '@chakra-ui/react'
+import DOMPurify from 'dompurify'
+
+/**
+ * Image with text component
+ *
+ * @param ITCLink Image Link.
+ * @param ITCText Text Below Image.
+ * @param image Image.
+ * @param heading Text Overlay.
+ * @param alt The image alt text shown by the component.
+ * @returns {JSX.Element}
+ */
+const ImageWithText = ({ITCLink, ITCText, image, heading, alt}) => {
+    const [sanitizedHtml, setSainitizedHtml] = useState(null)
+
+    useEffect(() => {
+        const purify = DOMPurify(window)
+        const cleanHeading = purify.sanitize(heading)
+        const cleanITCText = purify.sanitize(ITCText)
+        setSainitizedHtml({heading: cleanHeading, ITCText: cleanITCText})
+    }, [])
+
+    return (
+        <div className={'image-with-text'}>
+            <Box
+                as="figure"
+                className={'image-with-text-figure'}
+                position={'relative'}
+                margin={0}
+                width={'100%'}
+            >
+                <picture>
+                    <source srcSet={image?.src?.tablet} media="(min-width: 48em)" />
+                    <source srcSet={image?.src?.desktop} media="(min-width: 64em)" />
+                    <Link to={ITCLink}>
+                        <Image
+                            className={'image-with-text-image'}
+                            data-testid={'image-with-text-image'}
+                            src={image?.src?.mobile ? image?.src?.mobile : image?.url}
+                            ignoreFallback={true}
+                            alt={alt}
+                            title={alt}
+                            filter={'brightness(40%)'}
+                        />
+                    </Link>
+                </picture>
+                {(ITCText || heading) && (
```

Comment:
I see that compared to the ISML template we are being more smart about what html is rendered and what isn't based on the variables. That is good, but this line in particular isn't completely obvious as to what is going on, I think it's because of that fact that the one prop is uppercase making this look weird. 
A simple way to make this readable would be to do something like 

```
const hasCaption = ITCText || heading
```

Then use that variable in the template portion to make it easier to read.

Reply:
Agree. It's easier to read.